### PR TITLE
ARROW-14777: [Release] Enable to run on RHEL derivatives

### PR DESCRIPTION
**Pull request by bkmgit**

### CODE REVIEW

File: dev/release/02-source.sh [hunk 1]
```diff
@@ -85,11 +85,20 @@ if [ ${SOURCE_RAT} -gt 0 ]; then
   "${SOURCE_DIR}/run-rat.sh" ${tarball}
 fi
 
+if type shasum >/dev/null 2>&1; then
+  get_sha256="shasum -a 256"
```
Comment (kou) [inline dev/release/02-source.sh:89]: How about `sha256_generate`?

Comment (bkmgit) [reply dev/release/02-source.sh:89]: ok

File: dev/release/02-source.sh [hunk 2]
```diff
@@ -85,11 +85,20 @@ if [ ${SOURCE_RAT} -gt 0 ]; then
...
+fi
+
+
 if [ ${SOURCE_UPLOAD} -gt 0 ]; then
   # sign the archive
   gpg --armor --output ${tarball}.asc --detach-sig ${tarball}
-  shasum -a 256 $tarball > ${tarball}.sha256
-  shasum -a 512 $tarball > ${tarball}.sha512
+  ${get_sha256} $tarball > ${tarball}.sha256
+  ${get_sha512} $tarball > ${tarball}.sha512
```
Comment (kou) [inline dev/release/02-source.sh:101]: Could you also update them?

Comment (bkmgit) [reply dev/release/02-source.sh]: Sorry, thanks for spotting that.

**Review by kou — APPROVED** [review]
+1

**Pull request closed without merging**